### ISSUE TITLE
Is there a best example of multithreading or WebApi?

### ISSUE THREAD

**Issue by threerog** [sstainba/Yolov8.Net#38]
Is `Yolov8.Net` object thread safe?
As is well known, it is very time-consuming to load the model for the first time. How to efficiently use `Yolov8.Net` when making concurrent calls in the form of multithreading or `WebApi`?
Can you provide some relevant examples or solutions?
For example, is it feasible to use `lock` or `singleton` patterns

**Comment by sstainba** [comment]
I didn't write it with that use in mind, so I can't guarantee it.  I wrote this as a way to use it with a WPF app and my GPU.  It already uses the parallel library for processing vectors.

And no, I don't have examples of it for web applications.

**Comment by threerog** [comment]
Do you always infer new objects and load the model to use them in practice, or do you use the same object for repeated inference

**Comment by sstainba** [comment]
I use the same object.  Loading the model, even on the GPU, is the most time intensive operation.
________________________________

**Comment by threerog** [comment]
Are you using the same object in a single threaded scenario?
No thread safety control applied to it?

**Comment by sstainba** [comment]
Yes.  Runs in a task, wrapped in a using statement.  It's only disguised when the task is cancelled
